proxy/proxy_test.go

```diff
@@ -2277,3 +2277,40 @@ func BenchmarkAccessLogEnablePrint(b *testing.B) {
 	benchmarkAccessLog(b, "enableAccessLog(1,200,3)", 200)
 }
 func BenchmarkAccessLogEnable(b *testing.B) { benchmarkAccessLog(b, "enableAccessLog(1,3)", 200) }
+
+type zeroReader struct{}
+
+func (zeroReader) Read(p []byte) (int, error) {
```

Comment:
Would the benchmark results be the same if the (pseudo)random data would be copied?

Reply:
I think it might be different because this loop is optimized into memset iirc. But since we compare with and without buffer pooling but copy the same data the difference comes from the pooling and not from the data copied.